README.md

```diff
@@ -42,6 +42,53 @@ If you are migrating a service to DBT PaaS, [GOV.UK PaaS to DBT PaaS Migration](
     poetry run pre-commit install
     ```
 
+### Platform Helper architecture
+
+`platform-helper` is split into the following layers:
+
+Commands (UI) -> Domains -> Providers
+
+Code written for `platform-helper` should adhere to the following architecture, Shown below.
+
+![platform-helper architecture](doc/platform-helper-architecture.png)
+
+#### Commands
+
+This is the (essentially) UI level of platform-helper
+
+We try to follow a noun/domain verb/action pattern, `platform-helper thing action`.
+
+E.g. `codebase` has the following commands (all to do with codebase):
+
+- `build`
+- `deploy`
+- `list`
+- `prepare`
+
+Each command has an associated `domain`. 
+
+There should be no business logic within the command as this is extracted to the domain layer.
+
+CLI arguments for a command are pulled in via [click](https://click.palletsprojects.com/en/stable/) and passed to the domain code along with any dependencies.
+
+#### Domains
+
+Domains are where the business logic for a given command lives.
+
+Each domain is a class.
+
+Any logged information (`click.secho`) from the `provider` level should live within the `domain` level.
+
+Any common or repeatable code should not be defined within the `domain` and should instead be extracted to a suitable `provider`.
+
+#### Providers
+
+Providers are groups of similar logic that are linked by the resource/tool/thing they use and not by what the result of there actions is.
```

Comment:
```suggestion
Providers are groups of similar logic that are linked by the resource/tool/thing they use rather than the result of their actions is.
```